openedx/features/discounts/docs/decisions/0001-determining-discount-location.rst

```diff
@@ -0,0 +1,78 @@
+
+   b. We will calculate applicability/eligibility in the LMS. All of the data that we need currently resides in 
+      the LMS. In order to get the information to make the determination in Ecommerce, we would need to ask LMS
+      for the data. If the ultimate destination of that data was the LMS (to decide whether to message the 
+      discount), we would have excess communication and more failure points. See below for a diagram of what we
+      don't want:
+      |LMS -asks for discount applicability------------------> Ecommerce
+      |LMS <-----------asks for data to dermine applicability- Ecommerce
+      |LMS -sends back data----------------------------------> Ecommerce
+      |LMS <-------------------------sends back applicability- Ecommerce
+
+#. **Communication of discount applicability**
+
+   a. Within the LMS, when we need to get the applicability of a discount for a user and course, we can just
+      import the function and call it.
+
+   b. Outside of the LMS, we need a way to get this information in a validated fashion. We have added a rest 
+      endpoint for that communication. Because we want to be able to use the information both on the front-end
+      and back-end without risk of the user creating their own discounts, we are sending it as a jwt, which is
+      signed. See C for more information on how specifically it will be used on the Ecommerce front-end.
+
+   c. We do not want to make a synchronous call from the Ecommerce backend to the LMS every time the basket 
+      page is loaded. See this decision record link for more information: 
+      https://github.com/edx/Ecommerce/blob/master/docs/decisions/0002-no-synchronous-calls.rst  However, 
+      we are happy to make an ajax call from the front-end.
+      The basket page needs this information for two reasons: 1) So that the correct price is shown to the user,
+      and the user is informed that they are recieving a discount. 2) So that the correct price is charged to 
+      the user during payment.  We will handle both by making the ajax call, updating the ui on the page, and
+      also updating the data that is sent to the server during the payment step. Because we are using a JWT, we
+      can verify that the discount is correct. Right now, another team is working on adding a MFE to Ecommerce. 
+      Rather than have to implement the front-end code in Ecommerce to do this twice, we will write the code
+      to make the ajax call, and if there is a discount, reload the page passing the JWT as a parameter. See
+      diagram below for current (interum) state of the communication of the discount:
+      |LMS <--asks about discount applicability- Ecom Frontend                                      Ecom Backend
+      |LMS -sends applicability as JWT---------> Ecom Frontend
+      |-----IF APPLICABLE----
+      |LMS                                       Ecom Frontend -Reloads the page w/ JWT as param--> Ecom Backend
```

Comment:
As I understand it, the frontend will be using the info in the jwt without verifying its signature. Once the jwt is sent to the backend, the backend will verify it. This means that frontend will accept an incorrectly-signed jwt but the backend will not.

This might be worth calling out.

Reply:
All the frontend is doing is sending it to the backend. It's not actually using it.